propolis/src/vmm/machine.rs

```diff
@@ -476,7 +476,7 @@ impl<'a> MemCtx<'a> {
         }
     }
 
-    /// Returns the [lowest, highest] memory addresses in the space, inclusive.
+    /// Returns the [lowest, highest + 1) memory addresses in the space.
```

Comment:
We've generally shied away from using clopen ranges because of overflow issues at the highest addresses, though I don't think that's much of a concern because the physical address space is constrained to be well within the range of a u64. Regardless, it strikes me that the original comment was just wrong; that range wasn't inclusive. One wonders whether another fix might be using an inclusive range?

Reply:
(For completeness, I also looked at changing `offer_ram` and its callees so that they expect `end` to be the last valid address instead of the one-past-the-end address. That also solves the expectation mismatch, but it creates a lot of code of the form `end = start + (page_count * page_size) - 1`, which I didn't love--it seems like most callers' natural inclination is to deal with addresses that are multiples of 4K, and making them all adjust downward by a byte seemed like it'd be a pebble in the shoes of future maintainers of this code. But I could be persuaded otherwise.)